fog/overseer/server/tests/inactive_outstanding_key_idle_does_not_have_key.rs

```diff
@@ -29,178 +19,91 @@ const PORT_NUMBER: u16 = 8081;
+        utils::add_test_block(&mut ctx.ledger, &ctx.watcher, &mut ctx.rng);
     }
 
     // Stop the current active node. This should make this node's key
     // "outstanding" because at this point in time, there will be no active node
     // that uses this key to scan.
-    drop(node0);
+    nodes.pop_front();
 
-    // Give node0 time to stop. There's a gRPC bug that prevents threads from
+    // Give nodes[0] time to stop. There's a gRPC bug that prevents threads from
     // being joined automatically, but if we give it a second then it should
     // successfully join the threads.
-    std::thread::sleep(Duration::from_secs(2));
+    sleep(Duration::from_secs(2));
 
-    // Delete the state file for node0. This ensures that when it's restarted,
-    // it won't be automatically moved to the active state.
-    drop(std::fs::remove_file(&state_file_node_0));
+    // Change the ingress keys on the remaining nodes so that they're different
+    // than former node0's ingress key, which is the currently active key.
+    nodes[0].set_new_keys().unwrap();
+    nodes[1].set_new_keys().unwrap();
+    sleep(Duration::from_secs(5));
 
-    // Change the ingress keys on node1 and node2 so that they're different than
-    // node0's ingress key, which is the currently active key.
-    node1.set_new_keys().unwrap();
-    node2.set_new_keys().unwrap();
-    std::thread::sleep(Duration::from_secs(5));
-
-    let first_node1_ingress_key = node1.get_ingest_summary().get_ingress_pubkey().clone();
-    let first_node2_ingress_key = node2.get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node1_ingress_key = nodes[0].get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node2_ingress_key = nodes[1].get_ingest_summary().get_ingress_pubkey().clone();
 
     assert_ne!(original_ingress_key, first_node1_ingress_key);
     assert_ne!(original_ingress_key, first_node2_ingress_key);
     assert_ne!(first_node1_ingress_key, first_node2_ingress_key);
 
-    // Restart node0. This mimics what happens when our cloud infra provider
+    // Restart nodes[0]. This mimics what happens when our cloud infra provider
     // "brings back" a bounced node.
-    let (node0, _state_file_node_0, _client_listen_uri0) = utils::make_node(
-        0,
-        peer_indices.iter().cloned(),
-        recovery_db.clone(),
-        &watcher_path,
-        &ledger_db_path,
-        logger.clone(),
-    );
-    assert!(!node0.is_active());
+    nodes.push_front(ctx.make_node(0, 0..3));
 
     for _ in 0..11 {
-        utils::add_test_block(&mut ledger, &watcher, &mut rng);
+        utils::add_test_block(&mut ctx.ledger, &ctx.watcher, &mut ctx.rng);
     }
 
     // It takes overseer on average 10 seconds to realize that the node has
-    // gone down. Make it 15 seconds to be safe.
-    std::thread::sleep(Duration::from_secs(15));
+    // gone down. Make it 20 seconds to be safe.
+    sleep(Duration::from_secs(20));
```

Comment:
This got bumped form 15 to 20, are there any implications or false assumptions that should be followed up on another issue?